File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let lastNonZeroIndex = 0;
    for(let i = 0;i<nums.length;i++){
        if(nums[i] !== 0){
            [nums[i],nums[lastNonZeroIndex]] = [nums[lastNonZeroIndex],nums[i]];
            lastNonZeroIndex++;
        }
    }
    
};

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        // tc o(n)
        // sc o(1)

        int n = nums.size();
        int l = 0, r = 0;
        while(r<n){
            if(nums[r]!= 0){
                swap(nums[l],nums[r]);
                l++;
            }
            r++;
        }



        
    }
};